woocommerce/woocommerce-ios
# Fix store creation not started when logging in from the store creation CTA in the prologue

Closes: #11078


## Why


Before free trials/IAP were supported, the current `Create a New Store` CTA copy did not mention store creation. Thus, if the merchant logs in from this CTA, we intentionally show a store picker if their account is connected to Woo stores instead of starting the store creation flow. Now that this CTA is about store creation, logging in from this CTA should start the store creation flow whether the account has existing stores or not.

## How

In `AuthenticationManager`, `autoSelectStoreOrPresentStoreCreationFlow` was updated to `presentStoreCreationFlowInitiatedFromPrologue` to start the store creation flow if the merchant logs in from the `Create a New Store` CTA in the prologue.

## Testing instructions


- Log out from the app if needed
- On the prologue screen, tap `Create a New Store`
- Enter an email address associated with an existing WPCom account or select `Log In`
- Complete the login flow --> it should start the store creation flow

Regression testing:

- Log out from the app if needed
- On the prologue screen, tap `Log In`
- Enter a site address for a WPCOM site
- Enter the WPCOM credentials to log in --> it should log in to the store, not start the store creation flow

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/1945542/0dbb1113-1771-4c8b-aaf7-634bb63f4642



---
